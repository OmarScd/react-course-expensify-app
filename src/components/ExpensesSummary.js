import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>
    <p>
      Viewing&nbsp;
      {expenseCount}&nbsp;
      {expenseCount === 1 ? 'expense' : 'expenses'} totalling&nbsp;
      {numeral(expensesTotal / 100).format('$0,0.00')}
    </p>
  </div>
);

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: expenses.length,
    expensesTotal: getExpensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);