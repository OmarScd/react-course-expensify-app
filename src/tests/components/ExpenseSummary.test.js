import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly' , () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCout={2}
      expensesTotal={9434}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with one expense correctly' , () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCout={1}
      expensesTotal={9434}
    />
  );
  expect(wrapper).toMatchSnapshot();
});