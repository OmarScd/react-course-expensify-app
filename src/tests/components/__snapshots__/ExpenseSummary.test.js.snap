// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $94.34
  </h1>
</div>
`;

exports[`should render ExpensesSummary with one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $94.34
  </h1>
</div>
`;
